### DISCUSSION TITLE
How to install HPIPM and qpOASES for  use in qpsolvers?

### DISCUSSION THREAD

**Discussion by sandeep026** [qpsolvers/qpsolvers#247] [Q&A] [answered]
Dear developers,

I want to install hpipm and qpoases for use with qpsolvers. 
1. There has been instruction provided on the [qpsolvers website](https://qpsolvers.github.io/qpsolvers/installation.html) to install qpoases but on my system it takes forever.  
2. For hpipm,  I did not see any instructions. Can you guide me with the installation?
3. pip install "qpsolvers[open_source_solvers]" install all open source qp solvers from PIP. Can I install a list of specific solvers which the user provided?

```
conda install qpoases -c conda-forge        
Retrieving notices: ...working... done
Collecting package metadata (current_repodata.json): done
Solving environment: unsuccessful initial attempt using frozen solve. Retrying with flexible solve.
Solving environment: done
Collecting package metadata (repodata.json): done
Solving environment: unsuccessful initial attempt using frozen solve. Retrying with flexible solve.

CondaError: KeyboardInterrupt

^CTerminate batch job (Y/N)? 
^C
```

I am using 
1. windows 10
3. VScode
4.  python 3.11.5

**Comment by stephane-caron** [comment, marked as answer]
For your first two questions: we don't provide support for installing individual solvers, but you should be able to ask your questions in the issue trackers of the solvers themselves:

* For qpOASES: you can ask your question in the [qpOASES issue tracker](https://github.com/coin-or/qpOASES/issues).
* For HPIPM: you can find instructions in the [HPIPM readme](https://github.com/giaf/hpipm/#python), and report any issue you run into to the [HPIPM issue tracker](https://github.com/giaf/hpipm/issues).

Your third point makes sense and is easy to set up :+1: → working on it in https://github.com/qpsolvers/qpsolvers/pull/248

**Reply by stephane-caron** [reply]
And it is merged and released :package: Now we can do *e.g.* `pip install qpsolvers[clarabel,daqp,proxqp,scs]`.
